lib/src/method_channel/mobile_scanner_method_channel.dart

```diff
@@ -326,12 +326,22 @@ class MethodChannelMobileScanner extends MobileScannerPlatform {
 
     _textureId = textureId;
 
+    DeviceOrientation? initialDeviceOrientation;
+
     if (defaultTargetPlatform == TargetPlatform.android) {
       _surfaceProducerDelegate =
           AndroidSurfaceProducerDelegate.fromConfiguration(
             startResult,
             cameraDirection,
           );
+      initialDeviceOrientation =
+          _surfaceProducerDelegate?.initialDeviceOrientation;
+    } else if (defaultTargetPlatform == TargetPlatform.iOS ||
```

Comment:
Use a Map pattern with a when clause :)

```suggestion
    } else if (startResult case {'initialDeviceOrientation': final String orientation} when defaultTargetPlatform == TargetPlatform.iOS || defaultTargetPlatform == TargetPlatform.macOS) {
       initialDeviceOrientation = orientation.parseDeviceOrientation();
    }
```